homeworks/3/ChatApp/src/com/company/Client.java

```diff
@@ -0,0 +1,109 @@
+package com.company;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.Socket;
+
+public class Client implements Runnable{
+    private String ip ;
+    private int port ;
+    private boolean done;
+
+    private Socket client;
+    private BufferedReader in;
+    private PrintWriter out;
+
+
+    public void shutdown() {
+        done = true;
+        try{
+            in.close();
+            out.close();
+            if(!client.isClosed()){
+                client.close();
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void connect(){
+        BufferedReader inReader = new BufferedReader (new InputStreamReader (System. in));
+
+        String message = null;
+        try {
+            message = inReader.readLine();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        if(message.startsWith("/connect")){
+            String[] command = message.substring("/connect ".length()).split(":");
+
+            ip = command[0];
+
+            port = Integer.parseInt(command[1]);
+
+        }
+    }
+
+
+
+    @Override
+    public void run() {

```

Comment:
This can become the `main` method and `implements Runnable` can be removed from the class definition because the client machine should be running a single `Client` object and so there is no reason to be a thread.